### PR TITLE
Repro 0x11f - Requesting unknown blobs - offline 

### PR DESCRIPTION
**Pull request by tyler-cai-microsoft**
Parent bug: [AB#11708](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/11708)
Task: [AB#12481](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/12481)

This test repros 0x11f
Repro steps:
1. Start a promise creating a blob and then storing the blob handle.
2. Before the blob is successfully uploaded, serialize the container.
3. This should store the blob handle, and that op should be serialized.
4. load a remote container to listen to ops
5. load the serialized container
6. The serialized container sends the blob handle op first.
7. It should send the blob attach op second
8. On the remote container, before the blob attach op is received and processed, load the blob handle.
9. 0x11f will hit since the blob attach op had not yet been processed/received.

Note: if we wait a little longer and process the blob attach op (which can be the very next op), this issue does not occur.

### CODE REVIEW

File: packages/test/test-end-to-end-tests/src/test/data-virtualization/blobHandles.spec.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
+/*!
```
Comment (markfields) [inline packages/test/test-end-to-end-tests/src/test/data-virtualization/blobHandles.spec.ts:1]: Since it's not DV-related, move to a different file?

**Comment by markfields** [comment]
So the bug is that lines 6 and 7 should be reversed, that we would somehow guarantee that the blob attach op is submitted before the blob reference is added, right?

File: packages/test/test-end-to-end-tests/src/test/data-virtualization/blobHandles.spec.ts (hunk 1 above)
Comment (tyler-cai-microsoft) [reply packages/test/test-end-to-end-tests/src/test/data-virtualization/blobHandles.spec.ts:1]: 100%

**Comment by tyler-cai-microsoft** [comment]
> So the bug is that lines 6 and 7 should be reversed, that we would somehow guarantee that the blob attach op is submitted before the blob reference is added, right?

I'd rather not jump to solutions, as there was a reason why the code was written as it was. This is a rather hard to reach race condition, and there are some performance implications to some of the solutions we might want to offer.

**Comment by rajatch-ff** [comment]
I believe Mark is referring to step 6 and 7 in your PR Description

---
In reply to: [2278821255](https://github.com/microsoft/FluidFramework/pull/22169#issuecomment-2278821255) [](http://example.com/codeflow?ancestors=2278821255)

File: packages/test/test-end-to-end-tests/src/test/offline/blobHandles.spec.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+							console.log("Throwing error");
+							throw error;
+						}
+					});
+				});
+			},
+		);
+	});
+
+	it("Should have blob when handle op is sequenced", async () => {
```
Comment (rajatch-ff) [inline packages/test/test-end-to-end-tests/src/test/offline/blobHandles.spec.ts:105]: >it [](http://example.com/codeflow?start=1&length=2)

can we annotate this multi-step test, possibly leveraging what you already have in PR Description. Helps readability.

**Review by rajatch5 — APPROVED** [review]
:shipit:

**Pull request merged**